lightly/utils/dependency.py

```diff
@@ -0,0 +1,27 @@
+import functools
+
+
+@functools.lru_cache(maxsize=1)
+def torchvision_vit_available() -> bool:
+    try:
+        # Requires torchvision >=0.12
+        import torchvision.models.vision_transformer
+    except (
+        RuntimeError,  # Different CUDA versions for torch and torchvision
+        OSError,  # Different CUDA versions for torch and torchvision (old)
+        ImportError,  # No installation or old version of torchvision
+    ):
+        return False
+    else:
+        return True
+
+
+@functools.lru_cache(maxsize=1)
+def timm_vit_available() -> bool:
+    try:
+        # Requires timm >= 0.9.9
+        import timm.models.vision_transformer
```

Comment:
This import has existed in timm since v0.1, so this won't actually test whether or not newer timm versions are installed. The specific import that failed for us in TorchGeo was `timm.layers.LayerType`, which was added in 0.9.9. I _think_ that's the newest feature you're using.

Reply:
Thanks! I added a check for `LayerType` specifically. With timm 0.9.8 we now indeed skip importing the MaskedCausalVisionTransformer and with timm 0.9.9 we import it.